Objective-C/CBLDatabase.mm

```diff
@@ -624,9 +655,30 @@ - (nullable NSDate*) getDocumentExpirationWithID: (NSString*)documentID {
 #pragma mark - INTERNAL
 
 - (void) mustBeOpen {
-    if (_c4db == nullptr) {
+    if ([self isClosed])
         [NSException raise: NSInternalInconsistencyException
                     format: @"%@", kCBLErrorMessageDBClosed];
+}
+
+- (void) mustBeOpenLocked {
+    CBL_LOCK(self) {
+        [self mustBeOpen];
+    }
+}
+
+- (void) mustBeOpenAndNotClosing {
```

Comment:
probably one line of comment saying, call this function only from the lock, since `_isClosing` is getting accessed.

Reply:
It's sort of having naming conversion around those methods already but I could add a comment.

Reply:
I updated the comments. Thanks.